cdap-ui/app/cdap/components/DataPrep/Directives/MapToTarget/index.tsx

```diff
@@ -0,0 +1,220 @@
+
+const useStyles = makeStyles({
+  clearfix: {
+    '&::after': {
+      display: 'block',
+      clear: 'both',
+      content: 'none',
+    },
+  },
+  floatRight: {
+    float: 'right',
+  },
+});
+
+interface IMapToTargetProps {
+  isOpen: boolean;
+  isDisabled: boolean;
+  column: string;
+  onComplete: () => void;
+  close: () => void;
+  dataModelList?: IDataModel[];
+  targetDataModel?: IDataModel;
+  targetModel?: IModel;
+}
+
+const MapToTarget = (props: IMapToTargetProps) => {
+  const classes = useStyles(undefined);
+
+  const {
+    isOpen,
+    isDisabled,
+    column,
+    onComplete,
+    close,
+    dataModelList,
+    targetDataModel,
+    targetModel,
+  } = props;
+
+  const [loadingText, setLoadingText] = useState('');
+  const [searchText, setSearchText] = useState('');
+
+  useEffect(() => {
+    let pending = true;
+    setLoadingText(`${PREFIX}.initializingText`);
+
+    const initialize = async () => {
+      try {
+        await loadTargetDataModelStates();
+      } catch (error) {
+        setError(error, 'Map to Target directive initialization failed');
+      } finally {
+        if (pending) {
+          setLoadingText('');
+        }
+      }
+    };
+
+    initialize();
+
+    return () => {
+      pending = false;
+    };
+  }, []);
+
+  useEffect(() => {
+    if (isOpen && !isDisabled) {
+      setPopoverOffset(document.getElementById('map-to-target-directive'));
+    }
+  });
+
+  const selectTargetDataModel = async (dataModel: IDataModel) => {
+    setLoadingText(`${PREFIX}.loadingText`);
+    try {
+      setTargetModel(null);
+      await setTargetDataModel(dataModel);
+      setSearchText('');
+    } catch (error) {
+      setError(error, 'Could not set target data model');
+    } finally {
+      setLoadingText('');
+    }
+  };
+
+  const selectTargetModel = (model: IModel) => {
+    setTargetModel(model);
+    setSearchText('');
+  };
+
+  const applyDirective = async (field: IModelField) => {
+    setLoadingText(`${PREFIX}.executingDirectiveText`);
+    try {
+      await saveTargetDataModelStates();
+    } catch (error) {
+      setError(error, 'Could not save target data model states');
+      setLoadingText('');
+      return;
+    }
+
+    executeDirective(field);
+  };
+
+  const executeDirective = (field: IModelField) => {
+    const directive =
+      'data-model-map-column ' +
+      `'${targetDataModel.url}' '${targetDataModel.id}' ${targetDataModel.revision} ` +
+      `'${targetModel.id}' '${field.id}' :${column}`;
+
+    execute([directive], false, true).subscribe(
+      () => {
+        close();
+        onComplete();
+      },
+      (error) => {
+        setError(error, 'Error executing Map to Target directive');
+      }
+    );
+  };
+
+  const renderDetail = () => {
+    if (!isOpen || isDisabled) {
+      return null;
+    }
+
+    return (
+      <div className="second-level-popover" style={{ padding: 0 }} onClick={preventPropagation}>
```

Comment:
Can we please avoid inline styles. Can we see if we can increase the specificity of this styling?

Reply:
I know that it is not good idea. The problem is that `.dataprep-columns-action-dropdown .action-item .second-level-popover` combination defines padding of `10px`. Because of that items highlighting does not look as in other menus (for example, **Format** directive). So I need to override this padding and reset it to 0. Another directives solve it in SCSS. I can't find how to do that in JSS except `!important` option that @Biao didn't accept previously.

Reply:
I am not sure if I understand. If we are trying to make things consistent why are we overriding second level popover padding here? We are rendering `CurrentSelection`, `Options` and `OptionList` with padding to the container. Can we remove those? This is just based on looking at the code, I am unable to pull the changes and try out locally otherwise can provide a very concrete feedback.

Reply:
Let me show you, for example, how **Format** directive looks like:
![format-directive](https://user-images.githubusercontent.com/50941274/82177377-6ebc3080-9903-11ea-941e-c06c1cd1e8c5.png)
As you can see highlighted item takes all width of the popover.
But popover defines padding of `10px` for its content and Format directive overrides it:
![format-directive-style](https://user-images.githubusercontent.com/50941274/82177442-ae831800-9903-11ea-896d-1e3bd9541220.png)
Lets see how **Map to Target** directive looks like when we do not override this padding:
![map-to-target-directive](https://user-images.githubusercontent.com/50941274/82177517-de322000-9903-11ea-87c2-d9f6ca5b959d.png)
Green border around is that padding from popover style.
In that case highlighted item will look as:
![map-to-target-directive-style](https://user-images.githubusercontent.com/50941274/82177626-484ac500-9904-11ea-98b7-b38262b8d490.png)
So the highlighted item does not take all the width because of popover padding.

If it is fine with you I can remove all the paddings I added and style `style={{ padding: 0 }}`.
Another option I can remove it as I said and add style to `OptionList`:
![option-list-style](https://user-images.githubusercontent.com/50941274/82177841-e2ab0880-9904-11ea-9338-796131a53a81.png)
In that case it will look like this:
![option-list](https://user-images.githubusercontent.com/50941274/82177972-26057700-9905-11ea-86ff-4dd1b86f700d.png)

What do you think?

Reply:
I think the second option looks good. We can give a negative margin for options list to cancel out the padding for second level popover. Thanks for taking the time to explain it.

Reply:
Done.